src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/DoNotDeclareStaticMembersOnGenericTypesTests.cs

```diff
@@ -428,6 +428,21 @@ public override int GetHashCode()
 ");
         }
 
+        [Fact, WorkItem(7126, "https://github.com/dotnet/roslyn-analyzers/issues/7126")]
+        public async Task CSharp_CA1000_ShouldNotGenerate_VirtualMember()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+#if NET7_OR_GREATER
```

Comment:
Can we condition the entire test method on `NET7_OR_GREATER`?

Reply:
It turns out the `#if` was causing nothing to be generated (since the .NET version we're testing with is less than 7).

So I specified C# 11 and added expected diagnostics about .NET 7 features. It doesn't look very pleasant, if there's a more elegant way I'd be happy to improve things.

Reply:
OK I figured out how to specify .NET 6. Thank you for bearing with my learning, I'm fairly new to this codebase.